aplib.net-demo/Assets/Scripts/WFC/GridPlacer.cs

```diff
@@ -160,10 +172,59 @@ private void PlaceTile(int x, int z, Tile tile)
                 _ => _roomObjects.Empty
             };
 
-            tile.GameObject = Instantiate(prefab,
+            tile.GameObject = Instantiate
+            (
+                prefab,
                 new Vector3(x * _tileSizeX, 0, z * _tileSizeZ),
                 Quaternion.Euler(0, tile.Facing.RotationDegrees(), 0),
-                transform);
+                transform
+            );
+        }
+
+        /// <summary>
+        /// Place the doors for the given room in the world. Which doors need to be spawned is determined from the
+        /// allowed directions of the room.
+        /// </summary>
+        /// <param name="x">The x-position of the room, in the grid.</param>
+        /// <param name="z">The z-position of the room, in the grid.</param>
+        /// <param name="room">The room for which the doors need to be spawned.</param>
+        // ReSharper disable once SuggestBaseTypeForParameter
+        private void PlaceDoors(int x, int z, Room room)
+        {
+            Vector3 roomPosition = new(x * _tileSizeX, 0, z * _tileSizeZ);
+            // Get (half of) the depth of the door model
+            float doorDepthExtend = _doorRenderer.bounds.extents.z;
+            // Calculate the distance from the room center to where a door should be placed
+            float doorDistanceFromRoomCenter = (_tileSizeX / 2f) - doorDepthExtend;
+
+            Quaternion roomRotation = Quaternion.Euler(0, room.Facing.RotationDegrees(), 0);
+
+            foreach (Direction direction in room.ConnectingDirections)
+            {
+                // # Calculate where the door should be placed
+
+                // North is in the negative x direction
+                Vector3 relativeDoorPosition = direction switch
+                {
+                    North => new Vector3(-doorDistanceFromRoomCenter, 0, 0),
+                    East => new Vector3(0, 0, doorDistanceFromRoomCenter),
+                    South => new Vector3(doorDistanceFromRoomCenter, 0, 0),
+                    West => new Vector3(0, 0, -doorDistanceFromRoomCenter),
+                    _ => throw new UnityException("Invalid direction when placing door")
+                };
+                Vector3 doorPosition = roomPosition + relativeDoorPosition;
+
+                // # Calculate the rotation the door should have
+
+                // The `RotateLeft` here is because the rotation of the grid and the rotation of the door model do not
+                // line up
```

Comment:
Nit
```suggestion
                // # Calculate the rotation the door should have.

                // The `RotateLeft` here is because the rotation of the grid and the rotation of the door model do not
                // line up.
```